### ISSUE TITLE
[REVIEW]: Artery.FE: An implementation of the 1D blood flow equations in FEniCS

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1107]
**Submitting author:** @akdiem (<a href="http://orcid.org/0000-0003-1719-1942">Alexandra Diem</a>)
**Repository:** <a href="https://github.com/KVSlab/bloodflow" target ="_blank">https://github.com/KVSlab/bloodflow</a>
**Version:** v1.0
**Editor:** @trallard
**Reviewer:** @dlagrava, @jhill1
**Archive:** <a href="https://doi.org/10.5281/zenodo.2383815" target="_blank">10.5281/zenodo.2383815</a>

## Status

[![status](http://joss.theoj.org/papers/bfaec3f309adc6e836eb793dd652a23c/status.svg)](http://joss.theoj.org/papers/bfaec3f309adc6e836eb793dd652a23c)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/bfaec3f309adc6e836eb793dd652a23c"><img src="http://joss.theoj.org/papers/bfaec3f309adc6e836eb793dd652a23c/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/bfaec3f309adc6e836eb793dd652a23c/status.svg)](http://joss.theoj.org/papers/bfaec3f309adc6e836eb793dd652a23c)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/KVSlab/bloodflow" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@dlagrava & @jhill1, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.theoj.org/about#reviewer_guidelines. Any questions/concerns please let @trallard know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/KVSlab/bloodflow">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** Does the release version given match the GitHub release (v1.0)?
- [x] **Authorship:** Has the submitting author (@akdiem) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/KVSlab/bloodflow">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** Does the release version given match the GitHub release (v1.0)?
- [x] **Authorship:** Has the submitting author (@akdiem) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @dlagrava, it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01107/joss.01107/10.21105.joss.01107.pdf)

**Comment by trallard** [comment]
Hi reviewers and @akdiem ! This is the issue we will be using to conduct the review. I know @jhill1 is away for a couple of weeks but in the meantime, @dlagrava can get started with the review and recommendations (if any).

If any of you encounters any issues / need any guidance during the review process feel free to ping me here 👩🏻‍💻

**Comment by dlagrava** [comment]
@akdiem I have started this. The first comment that I have is that you actually do not have a release v1.0 on your github. Could you please create one? Thanks.

**Comment by akdiem** [comment]
@dlagrava Apologies, I must have forgotten because it is not mentioned in the submission guide. I've done this now.

**Comment by trallard** [comment]
Thanks for the note @akdiem I will revise the submission guide and make this clearer for future releases

**Comment by dlagrava** [comment]
@akdiem I have gone through the installation part. It is really a great idea to provide a Docker container. My only comment at this point is that the `python` command inside the container points to python 2, and you need to specifically need to execute the code with `python3`. Could you please update the readthedocs to reflect this or make the default python to be 3 on the Dockerfile. Thanks.

**Comment by dlagrava** [comment]
@akdiem I have tried to run the unit tests that you have provided. The `test_artery_network.py` does not work as the class it reffers to seems to have changed from "Artery_Network" to "ArteryNetwork". Thanks.

**Comment by akdiem** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01107/joss.01107/10.21105.joss.01107.pdf)

**Comment by akdiem** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01107/joss.01107/10.21105.joss.01107.pdf)

**Comment by jhill1** [comment]
Hi @akdiem. I found the package very interesting and nice use of Fenics. The module is very well documented with inline comments, and the documentation has examples, tests. The paper is well structured with a clear aim for the software and the equation laid out are easy to find in the corresponding code. The tests worked out of the box for me and installation was straightforward using the Docker or the more traditional setup.py.

The example given is well described, giving reference to the paper that it uses and the plot produced even use a good colour scheme!

There are a few minor issues with formatting of references in the paper and I would recommend hooking up the tests to TravisCI or similar system for continuous testing. The former should be done before accepting this paper, but the latter is at the discretion of the authors (as there are manually running tests).

Once the issues with the referencing are sorted, this paper should be accepted.

Note to @trallard I'm not an expert in blood flow equations, but the Fenics code looks good to me.

**Comment by dlagrava** [comment]
Hello @akdiem I have finished my review as well. I performed some blood simulations for stent optimisation on 3D and it was very time consuming, probably a 1D approach should be more efficient, while retaining most of the average behavior. I quite enjoyed the presentation of the theory of the 1D bloodflow equations on your readthedocs, which is a nice touch for the interested reader.

Your implementation based on FEnICS is well  documented, the installation and the tests work well. One question would be how you manage/inform about updates and bug fixes on FEnICS.

The last comment that I have is that your readme.md on the project should provide guidelines stating that people are welcome to contribute and submit bugs, and who to ask questions to. 

It was an interesting review. I wish you a lot of success.

@trallard Once the comment about community guidelines is fixed, I have finished with the review.

**Comment by akdiem** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01107/joss.01107/10.21105.joss.01107.pdf)

**Comment by akdiem** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01107/joss.01107/10.21105.joss.01107.pdf)

**Comment by akdiem** [comment]
@jhill1 Thank you for your review and your comments! I agree that CI use would be ideal. It has been quite tricky so far to set up the specific version of Fenics on Travis as I would need to install it via docker. I will look into this again though.

**Comment by akdiem** [comment]
@dlagrava Thank you for your review and your comments! I have added a "Support" section to the README, stating the info you requested. If you would like to explore running your simulations using artery.FE, I'd be more than happy to assist. Feel free to contact me about this.

**Comment by dlagrava** [comment]
@trallard sorry, that was an oversight. It is fixed now. I did finish the review. Thanks a lot!

**Comment by trallard** [comment]
Hey no worries @dlagrava thanks again for your time

As soon as @jhill1 confirms that the community guidelines point has been satisfactorily added I will conduct the last editorial tasks on the submission

**Comment by jhill1** [comment]
Done!

**Comment by trallard** [comment]
@akdiem now that the reviews are completed and the paper has been recommended for acceptance, we need you to complete the following actions:

- [ ]   make a new release of the package (unless the latest release is up to date with the recent modifications)

- [ ]  upload the revised software to your DOI-granting data/software repository, and post the DOI here

**Comment by akdiem** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01107/joss.01107/10.21105.joss.01107.pdf)

**Comment by akdiem** [comment]
@trallard I had forgotten to add the acknowledgements to the paper - I hope it is ok to still add them. I have rebuilt the paper and will perform the other final steps now.

**Comment by trallard** [comment]
@whedon set 10.5281/zenodo.2383815 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.2383815" target="_blank">10.5281/zenodo.2383815</a> is the archive.

**Comment by trallard** [comment]
@arfon: this submission is accepted and ready to be published 🎉👾

@jhill1  and @dlagrava  thank you very much for your time and valuable contribution to JOSS as a reviewer for this submission 🙌🏻🌟

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/123

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/123, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/124
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01107](https://doi.org/10.21105/joss.01107)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by arfon** [comment]
@dlagrava, @jhill1 - many thanks for your reviews here and to @trallard for editing this submission ✨ 

@akdiem - your paper is now accepted into JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](http://joss.theoj.org/papers/10.21105/joss.01107/status.svg)](https://doi.org/10.21105/joss.01107)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01107">
  <img src="http://joss.theoj.org/papers/10.21105/joss.01107/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: http://joss.theoj.org/papers/10.21105/joss.01107/status.svg
   :target: https://doi.org/10.21105/joss.01107
```
This is how it will look in your documentation:

[![DOI](http://joss.theoj.org/papers/10.21105/joss.01107/status.svg)](https://doi.org/10.21105/joss.01107)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: http://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://www.flipcause.com/secure/cause_pdetails/Mjk3Nzk=
